### PR TITLE
Add person,student and teacher classes

### PR DESCRIPTION
**Pull request by Kakalanp**
In this feature we're adding three classes with their own methods:
- Person:
  - is_of_age? (private)
  - can_use_services? (public)
- Student (inherits from Person):
  - play_hooky
- Teacher (also inherits from Person):
  - can_use_services? (overrides)

### CODE REVIEW

File: person_class.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    @age = age
+    @parent_permission = parent_permission
+    @id = Random.rand(1..1000)
+  end
+
+  private
+
+  def of_age?
+    p @age >= 18
+  end
```
Comment (ShoiraTa) [inline person_class.rb]: - [x] It looks like you have a typo here,  `p` isn't supposed to be there 😉

File: student_class.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require './person_class'
+
+Class Student < Person
+def initialize(classroom)
+  super
+  @classroom = classroom
+end
```
Comment (ShoiraTa) [inline student_class.rb]: - [x] Although Student extends Person, you have to initialise Student by necessary params:
```ruby
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
end

File: teacher_class.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require './person_class'
+
+Class Teacher < Person
+def initialize(specialization)
+  super
+  @specialization = specialization
+end
```
Comment (ShoiraTa) [inline teacher_class.rb]: - [x] Same goes here, kindly initialyze params:
```ruby
  def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @Kakalanp ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Using individual `.rb` file for each class ✔️ 
- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person_class.rb (hunk 1 above)
Comment (Kakalanp) [reply person_class.rb]: Thank you, I'll fix it right away! Haha

File: student_class.rb (hunk 2 above)
Comment (Kakalanp) [reply student_class.rb]: Thanks, I didn't know that!

**Review by Kingstalux — APPROVED** [review]
Hi @Kakalanp,
# Status: Approved ✔️ ✔️ ✔️ 
 
128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c


Good job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kakalanp** [comment]
Thank you very much! Merging right away

**Pull request merged**